luigi/contrib/slack_notifications.py

```diff
@@ -0,0 +1,148 @@
+
+'''
+Supports sending slack notifications for different task events.
+
+Usage:
+
+Create an incomming webhook on slack: https://api.slack.com/incoming-webhooks
+
+Use a configuration like the following on luigi.cfg
+.. code-block:: ini
+    [slack]
+    webhook: https://hooks.slack.com/services/...
+    events: ["START", "SUCCESS", "FAILURE"]
+    template: Luigi task {task_id} event {event}: {args} {kwargs}
+
+
+In the python module, setup the slack integration with:
+.. code-block:: python
+    from luigi.contrib.slack_notifications import configure_slack
+    configure_slack()
```

Comment:
Yes. I am thinking how to improve this wording.

This is where I mostly unsure, and I would like to explain my reasoning. This code needs to execute on luigi initialization or before the tasks run. I really did not find a nice place for this to execute and for now I am recommending to use leave at the end of the python module.

Alternatively, I would like call this at the `interface.py` or `worker.py`, as that does not have any side effect if slack is not configured. But then I think it is too much change.

The goal here is not to create a new `luigi_cmd` wrapping `luigi.run` (as https://github.com/bonzanini/luigi-slack) and also calling `configure_slack()`.